pkgs/applications/science/electronics/kicad/default.nix

```diff
@@ -35,17 +35,16 @@ let
   python = python3;
   wxPython = python3Packages.wxPython_4_0;
 
-  libraries = callPackages ./libraries.nix versionConfig.libVersion;
+in
+stdenv.mkDerivation rec {
+
+  passthru.libraries = callPackages ./libraries.nix versionConfig.libVersion;
```

Comment:
I don't think this is going to work. My understanding is that "nix-env" does not recurse into a derivation as if it was an attrset, neither will our release jobset (see https://github.com/NixOS/nixpkgs/pull/82760#issue-389547299 on how I was checking it there).

Reply:
I mention this because later you add `hydraPlatforms` to `mkLib`. Otherwise, I think exposing libraries, by itself, is a step in the right direction towards making things more overridable.

Reply:
i added `hydraPlatforms` to `mkLib` to guard against attempting to build `packages3d` on hydra in case it's ever figured out to make the components build on their own.
i'm not sure how `nix-env` relates to this.
your (approx.) line from 82760: `echo ":p { inherit kicad-small kicad-unstable-small; }" | nix repl ./pkgs/top-level/release.nix`
reveals wxwidgets-3.1.2 and 3.0.4 are not supported on `x86_64-darwin`, but otherwise seems to evaluate.

Reply:
i've got it evaluating without error for darwin now
i don't have access to a darwin machine, but i'll try the same configuration on nixos tomorrow
`wxPython_4_0` doesn't account for `wxGTK` being `wxmac` on darwin
`libngspice` is currently set to `platforms.linux` though it should support darwin

Reply:
I just wanted to make sure that you understand that release.nix will not discover `kicad.libraries.footprints` on its own and that none of our tools will check its `hydraPlatforms` for it either.

Regarding the darwin support, you, of course, don't need to implement it if you can't test or use it.